File: lib/go/internal/binary/Base64Encoder.go
//|
//|  Copyright The Telepact Authors
//|  SPDX-License-Identifier: Apache-2.0
//|

package binary

// Base64Encoder defines methods for encoding and decoding pseudo-JSON Telepact messages via Base64.
type Base64Encoder interface {
	Decode(message []any) ([]any, error)
	Encode(message []any) ([]any, error)
}

File: lib/py/telepact/internal/binary/Base64Encoder.py
#|
#|  Copyright The Telepact Authors
#|  SPDX-License-Identifier: Apache-2.0
#|

from abc import ABCMeta, abstractmethod


class Base64Encoder(metaclass=ABCMeta):

    @abstractmethod
    def decode(self, message: list[object]) -> list[object]:
        pass

    @abstractmethod
    def encode(self, message: list[object]) -> list[object]:
        pass
